becquerel/core/fitting.py

```diff
@@ -0,0 +1,859 @@
+
+    def guess(self, y, x=None, dx=None, center_ratio=0):
+        xspan = x[-1] - x[0]
+        mu = x[0] + xspan * center_ratio
+        return [
+            ('{}amp'.format(self.prefix), 'value', y[0] - y[-1]),
+            ('{}mu'.format(self.prefix), 'value', mu),
+            ('{}sigma'.format(self.prefix), 'expr', 'gauss_sigma'),
+        ]
+
+
+class GaussErfModel(Model):
+
+    def __init__(self, *args, **kwargs):
+        super(GaussErfModel, self).__init__(gausserf, *args, **kwargs)
+        self.set_param_hint(
+            '{}fwhm'.format(self.prefix),
+            expr='{} * {}sigma'.format(FWHM_SIG_RATIO, self.prefix))
+
+    def guess(self, y, x=None, dx=None, center_ratio=0.5, width_ratio=0.5,
+              amp_ratio=0.9):
+        assert center_ratio < 1, \
+            'Center mask ratio cannot exceed 1: {}'.format(center_ratio)
+        assert width_ratio < 1, \
+            'Width mask ratio cannot exceed 1: {}'.format(width_ratio)
+        if x is None:
+            x = np.arange(0, len(y))
+        if dx is None:
+            dx = np.ones_like(x)
+        xspan = x[-1] - x[0]
+        mu = x[0] + xspan * center_ratio
+        msk = ((x >= (mu - xspan * width_ratio)) &
+               (x <= mu + xspan * width_ratio))
+        amp = np.sum(y[msk]/dx[msk])
+        sigma = xspan * width_ratio / 10.
+        return [
+            ('{}ampgauss'.format(self.prefix), 'value', amp*amp_ratio),
+            ('{}ampgauss'.format(self.prefix), 'min', 0.0),
+            ('{}amperf'.format(self.prefix), 'value', amp*(1.0 - amp_ratio)),
+            ('{}amperf'.format(self.prefix), 'min', 0.0),
+            ('{}mu'.format(self.prefix), 'value', mu),
+            ('{}mu'.format(self.prefix), 'min', x[0]),
+            ('{}mu'.format(self.prefix), 'max', x[-1]),
+            ('{}sigma'.format(self.prefix), 'value', sigma),
+            ('{}sigma'.format(self.prefix), 'min', 0.0),
+        ]
+
+
+class ExpModel(Model):
+
+    def __init__(self, *args, **kwargs):
+        super(ExpModel, self).__init__(exp, *args, **kwargs)
+        self.set_param_hint('{}amp'.format(self.prefix), min=0.)
+        self.set_param_hint('{}lam'.format(self.prefix), max=0.)
+
+    def guess(self, y, x=None, dx=None, num=1):
+        xl, yl = _xy_left(y, x=x, num=num)
+        xr, yr = _xy_right(y, x=x, num=num)
+        # TODO: update this hardcoded zero offset
+        lam = (xl - xr) / np.log(yl / (yr + 0.0001))
+        amp = yl / np.exp(xl / lam)
+        return [
+            ('{}lam'.format(self.prefix), 'value', lam),
+            ('{}lam'.format(self.prefix), 'max', -1e-3),
+            ('{}amp'.format(self.prefix), 'value', amp),
+            ('{}amp'.format(self.prefix), 'min', 0.0),
+        ]
+
+
+class ExpGaussModel(Model):
+    """A model of an Exponentially modified Gaussian distribution
+    (see https://en.wikipedia.org/wiki/Exponentially_modified_Gaussian_distribution)
+    """
+
+    fwhm_factor = 2*np.sqrt(2*np.log(2))
```

Comment:
This is defined already at the top as `FWHM_SIG_RATIO`

Reply:
Please do that

Reply:
Actually, it's not even called. Deleted!